TwilioStore.Services/NotificationService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Twilio;
+using TwilioStore.Interfaces.Exceptions;
+using TwilioStore.Interfaces.Services;
+
+namespace TwilioStore.Services
+{
+    public class NotificationService : INotificationService
+    {
+        private readonly TwilioRestClient _client;
+        private readonly INotificationConfiguration _config;
+
+        public NotificationService(INotificationConfiguration config = null)
+        {
+            _config = config ?? new NotificationConfiguration();
```

Comment:
I like this approach, nonetheless we could replace this conditional with polymorphism. I have in mind something like this:

```csharp
public NotificationService() : this(new NotificationConfiguration())
{
}

public NotificationService(INotificationConfiguration config)
{
    _config = config;
    _client = new TwilioRestClient(_config.AccountSid, _config.AuthToken);
}
```

I'd love to hear your thoughts on this. I tend to think that using polymorphism is a more OO approach. 😉